convert2rhel/pkghandler.py

```diff
@@ -301,8 +301,45 @@ def get_pkg_fingerprint(pkg_obj):
         return "none"
 
 
+@utils.run_as_child_process
+def get_pkg_signature_with_cleanup(pkg_obj):
+    """
+    Wrapper function for get_pkg_signature that will run as child and handle
+    rpmdb termination.
+
+    .. warning::
+        This function will run as a child process for the main thread, the
+        reason for that is that this was the only way to make it behave and
+        respect the SIGINT signals when the user presses Ctrl + C (SIGINT) or
+        send a `kill -<number> <pid>` to the main process. The reason that this
+        is done this way is because of RPM installing a global signal handler
+        during some functions calls that is propagated to the main thread
+        instead of being released after the function execution is over.
+
+        Running the function in a child process does not change how the
+        function itself works.
+
+        In addition to it running as a child process, this function will handle
+        the closing of the rpmdb instance to prevent further problems when
+        using Yum in a CLI, or, when there is a call to the Yum python API.
+
+    :param pkg_obj: Instace of a package RPM installed on the system.
+    :type pkg_obj: yum.rpmsack.RPMInstalledPackage
+    :return: Return the package signature.
+    :rtype: str
+    """
+    with pkgmanager.rpm_db_lock(pkg_obj):
+        return get_pkg_fingerprint(pkg_obj)
```

Comment:
Could we not use `get_pkg_signature()` here given the name of these two methods being nearly the same?

Reply:
`get_pkg_signature()` is called a lot of times sometimes, especially if we have a lot of packages to verify, and since we are closing the database by dropping all the cache (as well, as a child process), I created an wrapper function just to this specific part. 

Probably the name of the wrapper function could be improved, but not sure of what to call this.

Reply:
We should not change what we output in the breadcrumbs data without a good reason. With using `get_pkg_fingerprint()` instead of `get_pkg_signature()` you'd change the value of the `signature` breadcrumbs field from `kernel RSA/SHA256, Wed 15 Feb 2023 03:47:41 PM CET, Key ID f55ad3fb5323552a` to `f55ad3fb5323552a`.

```suggestion
    with pkgmanager.rpm_db_lock(pkg_obj):
	    return get_pkg_signature(pkg_obj)
```

Reply:
Good catch, @bocekm! I thought I was using the correct function in there, but it was the wrong one.